lib/std/Build.zig

```diff
@@ -1573,6 +1582,16 @@ pub fn dependency(b: *Build, name: []const u8, args: anytype) *Dependency {
         }
     }
 
+    inline for (@typeInfo(deps.build_root).Struct.decls) |decl| {
+        if (mem.startsWith(u8, decl.name, b.dep_prefix) and
+            mem.endsWith(u8, decl.name, name) and
+            decl.name.len == b.dep_prefix.len + name.len)
+        {
+            const build_root = @field(deps.build_root, decl.name);
+            return dependencyInner(b, name, build_root, null, args);
+        }
+    }
```

Comment:
sorry for not doing this myself, but could you add doc comments to Package.zig or to fetchAndAddDependencies to show examples of `dependencies_source` and `build_roots_source`? I'm not quite clear on why this new block of logic needed to be added.

Reply:
Dependencies without a build.zig file won't exist inside imports (because there is not build.zig to import), the second block instead looks through the build roots to find the root directory of the dependency. I'm adding a comment over this inline for and a doc comment to ```fetchAndAddDependencies```.